### ISSUE TITLE
Rendering "Stencil" color not working for 32009 UnderwaterLightRay anymore

### ISSUE THREAD

**Issue by Tormentor667** [Realm667/WolfenDoom#325]
I noticed that the rendering style "Stencil" or "Stencil Additive" with a color isn't working for the named actor anymore. It worked before and it works in my Sapphire remaster as well, just not in Blade of Agony anymore. Instead of white, the actors should be completely colored as described.

![2020-12-28_200242](https://user-images.githubusercontent.com/10298439/103237426-d92c5900-4947-11eb-832a-dac767dfb75c.jpg)
![2020-12-28_200319](https://user-images.githubusercontent.com/10298439/103237427-da5d8600-4947-11eb-96c3-5d6cd4cf5897.jpg)

Maybe this has something to do with certain refactors of visual special effect actors @Ozymandias81 @AFADoomer

**Comment by AFADoomer** [comment]
No idea what's going on here.  This actor hasn't been touched significantly in years...  I didn't change anything with the underwater rays for the effects revamp.

I've even tested pasting in the actor from Sapphire as a test, and it doesn't draw in color in BoA either...

**Comment by Tormentor667** [comment]
Is the actor inheriting any code that forces a different render style or color? 

I would have said something in GZDoom changed but I just tested Sapphire and it works there.

**Comment by AFADoomer** [comment]
No, just from SwitchableDecoration.  That's what doesn't make sense...  plus it renders correctly colored in GZDB, so I doubt it's the definition.

*Something* is affecting it in BoA, just not sure what.

**Comment by Tormentor667** [comment]
Has anything been added or changed in ZScript that changes dependencies on the SwitchableDecoration? Would be interesting to know if it works when the Ray doesn’t inherit from SwitchableDeco

**Comment by AFADoomer** [comment]
No, otherwise Sapphire wouldn't work either.

**Comment by Tormentor667** [comment]
Well, it works in Sapphire, but I thought the same actor from Sapphire doesn’t work in BoA either?

**Comment by AFADoomer** [comment]
Right.  SwitchableDecoration is a GZDoom base actor, so if it had changed, both would be broken.

**Comment by Tormentor667** [comment]
Just tried to remove the actor dependency from SwitchableDecoration. No change, the sprite is still white.

**Comment by Tormentor667** [comment]
Things are getting strange now. I duplicated the actor and assigned a different sprite to it in this commit, and voilá, the second actor with the different sprite works like a charme: https://github.com/Realm667/WolfenDoom/commit/20ad0b7df2223501e9e94098b193ba7b12563f00

What's strange is that I opened the graphic again in Photoshop and saved it as TrueColor, just as the other sprite, no success. Also tried in in greyscale, no success. If I use the VOLTC0 sprite for the first actor it works. Something seems to be having a problem with the sprite obviously.
